datalad/support/gitrepo.py

```diff
@@ -2388,34 +2388,34 @@ def get_submodules_(self, paths=None):
             # below
             return
 
-        # taken from 3.9's pathlib, can be removed once the minimally
-        # supported python version
-        def is_relative_to(self, *other):
-            """Return True if the path is relative to another path or False.
-            """
-            try:
-                self.relative_to(*other)
-                return True
-            except ValueError:
-                return False
-
+        posix_mod_paths = [m.relative_to(self.pathobj).as_posix() for m in modinfo]
```

Comment:
I think, this is a little weird. `get_submodules_` appears to be the only consumer of `_parse_gitmodules`. The latter transforms the recorded relative paths in `.gitmodules` into absolute ones, and the former transforms them back.

Reply:
> So, if nobody minds -- I should then just strip making them absolute within _parse_gitmodules.

No objection. :)